libres/lib/res_util/block_fs.cpp

```diff
@@ -318,22 +318,21 @@ static block_fs_type *block_fs_alloc_empty(const fs::path &mount_file,
 
     block_fs->fsync_interval = fsync_interval;
     block_fs->block_size = block_size;
-    {
```

Comment:
I had a reread parts of "clean code" last night so take this as a highly influenced opinion. Perhaps the block should be extracted as a function?

Reply:
Can the function be re-used you think?

Reply:
No, the style used by robert C. Martin is shorter functions (that do one thing at one abstraction level) where the name serves as documentation, ie.:
```cpp
static std::tuple<int, int> read_mount_file_header(const fs::path &mount_file){
    FILE *stream = util_fopen(mount_file.c_str(), "r");
    int id = util_fread_int(stream);
    int version = util_fread_int(stream);
    fclose(stream);
    return {id, version};
}

static void validate_mount_file(const fs::path &mount_file){
    auto &[id, version] = read_mount_file_header(mount_file);
    if (version != 0)
        throw std::runtime_error(fmt::format(
            "block_fs data version unexpected. Expected 0, got {}", version));
    if (id != MOUNT_MAP_MAGIC_INT)
        util_abort("%s: The file:%s does not seem to be a valid block_fs "
                   "mount map \n",
                   __func__, mount_file.c_str());
}

static block_fs_type *block_fs_alloc_empty(const fs::path &mount_file,
                                           int block_size, int fsync_interval,
                                           bool read_only) {
    block_fs_type *block_fs = new block_fs_type;
    UTIL_TYPE_ID_INIT(block_fs, BLOCK_FS_TYPE_ID);

    block_fs->fsync_interval = fsync_interval;
    block_fs->block_size = block_size;

    validate_mount_file(mount_file);

    block_fs_reinit(block_fs);

    block_fs->data_owner = !read_only;
    return block_fs;
}
```

Reply:
Also, should that second `util_abort` also be a `throw std::runtime_error`?